### PR TITLE
Make vmware_host idempotent

### PR DESCRIPTION
**Pull request by ahanwadi**

### CODE REVIEW

**Comment by robynbergeron** [comment]
Thanks @ahanwadi. @mtnbikenc @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by jcpowermac** [comment]
@ahanwadi I would rather see a revision to the `find_host_by_cluster_datacenter` method so that it properly checks the state if a host is available in vCenter or not.  If the host in vCenter is the same name as what is in `esxi_hostname` I am unsure why this would be happening.

**Comment by mtnbikenc** [comment]
@ahanwadi as @jcpowermac pointed out, the find_host_by_cluster_datacenter method is used to determine if the host already exists.  If there is a case in which this is not working, the method should be updated rather than handling it in the add_host_to_vcenter method.  Could you add more details to either the issue or this PR in regards to version of vSphere and version of Ansible being used?

needs_revision

**Comment by gregdek** [comment]
Thanks @ahanwadi for this PR. A maintainer of this module has asked for revisions to this PR. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by gregdek** [comment]
@ahanwadi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by chrrrles** [comment]
One way is to check if the `host.summary.managementServerIp` is set to the ip address of the vcenter server.  A complication is that values for the `hostname` module option may (obviously) be entered as DNS names, so the vcenter ip will have to be determined another way.

This fault may also be caused by a corrupt vcenter database, so more information would be appreciated.

**Comment by gregdek** [comment]
Thanks @ahanwadi for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by ahanwadi** [comment]
I rebased and pushed the new changes. The duplicateName fault is thrown when the same host is being added using its IP address and currently present in the vCenter inventory using its hostname.
Also, there is always a race condition between check for if already present and adding the host. The problem is whatever reason we get the fault other than the SSL verify fault, we end up accessing fields that are not defined in the fault parameter. So, we should fix this anyway. In fact, I would prefer handling SSL verify fault using a separate except clause.

**Comment by ahanwadi** [comment]
@gregdek I don't seem to have permission to apply a label.

**Comment by gregdek** [comment]
@ahanwadi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by gregdek** [comment]
@ahanwadi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ahanwadi** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @ahanwadi. To the current maintainers, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/vmware/vmware_host.py [hunk 1]
```diff
@@ -157,6 +157,11 @@ def add_host_to_vcenter(self):
             #
             # args is a tuple, selecting the first tuple
             ssl_verify_fault = add_task_error.args[0]
+            fault = add_task_error.args[0]
+            if type(fault) is vim.fault.DuplicateName:
+               return False, fault.object
+            ssl_verify_fault = add_task_error.args[0]
+            host_connect_spec.sslThumbprint = ssl_verify_fault.thumbprint
             host_connect_spec.sslThumbprint = ssl_verify_fault.thumbprint
```
Comment (resmo) [inline cloud/vmware/vmware_host.py:165]: I see 2 times the same line, can this be correct?

Comment (resmo) [inline cloud/vmware/vmware_host.py:165]: any updates about this? I

**Comment by resmo** [comment]
I made a code comment ans waiting for a reply. 

needs_info

**Comment by gregdek** [comment]
@ahanwadi, a community member requested some information about this PR; for that reason, we put this PR into the 'needs_info' state.

When you have answered the questions, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ahanwadi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ahanwadi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ahanwadi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Closing due to contributor inactivity.

**Comment by daBONDi** [comment]
Still an Issue, Module not usable :-(

**Pull request closed without merging**